### ISSUE TITLE
"mb_strtoupper() expects parameter 1 to be string" when Resource $title is a json field.

### ISSUE THREAD

**Issue by aniplaylist** [laravel/nova-issues#3042]
- Laravel Version: 8.14.0
- Nova Version: 3.15.0
- PHP Version: 7.4.6
- Database Driver & Version: Mysql 5.7
- Operating System and Version: Mac OS Catalina 10.15.6
- Browser type and version: Chrome 86.0.4240.111

### Description:

Just upgraded to from Nova 3.11.1 to 3.15.0 and I'm now getting the following error when trying to open detail or edit form of some of my resources.
```
"message": "mb_strtoupper() expects parameter 1 to be string, array given",
    "exception": "ErrorException",
    "file": "/var/www/html/vendor/laravel/framework/src/Illuminate/Support/Str.php",
    "line": 568,
    "trace": [...]
```
(I can provide the complete stack trace if needed)

After a bit of debugging I noticed that it was only happening for Resources for which I had defined `$title` to be a json column which is casted as array in my model.

### Detailed steps to reproduce the issue on a fresh Nova installation:

* Create `users` table with a json typed `names` column
* In your `User.php` model, cast the column as array
* In `Nova/User.php` define `names` column as $title : `public static $title = 'names';`
* Open detail or edit view for a User

### Workaround

I first tried another syntax such as `public static $title = 'names.0';` or `public static $title = 'names[0]';` but it didn't work. 

What I did in the end is override the resource's title method to make it work as I want.

I'm still filling this issue in case it can help improve the way `$title` behaves with json columns casted as array.

**Comment by davidhemphill** [comment]
Overriding the `title` is the right play here. 👍

**Comment by davidhemphill** [comment]
👋 Hi there! In order to keep this repository focused on bug reports, we auto-close feature requests and requests for help. Feel free to post your feature requests so others can discuss and add reactions. We'll keep an eye on them for later planning.

**Comment by crynobone** [comment]
Implemented and released on v3.21.1
